californium-core/src/main/java/org/eclipse/californium/core/network/stack/TcpAdaptionLayer.java

```diff
@@ -0,0 +1,80 @@
+/*******************************************************************************
+ * Copyright (c) 2015 Bosch Software Innovations GmbH and others.
+ * 
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ * 
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ * 
+ * Contributors:
+ *    Achim Kraus (Bosch Software Innovations GmbH) - Initial implementation
+ ******************************************************************************/
+package org.eclipse.californium.core.network.stack;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.eclipse.californium.core.coap.CoAP.Type;
+import org.eclipse.californium.core.coap.EmptyMessage;
+import org.eclipse.californium.core.coap.Request;
+import org.eclipse.californium.core.coap.Response;
+import org.eclipse.californium.core.network.Exchange;
+
+public class TcpAdaptionLayer extends AbstractLayer {
+
+	private static final Logger LOGGER = Logger.getLogger(TcpAdaptionLayer.class.getName());
+
+	public TcpAdaptionLayer() {
+	}
+
+	@Override
+	public void sendRequest(Exchange exchange, Request request) {
+		if (null == request.getType()) {
+			request.setType(Type.CON);
+		}
+		super.sendRequest(exchange, request);
+		// CoAP over TCP does not have acknowledgements. Everything is
+		// automatically acknowledged.
+		request.setAcknowledged(true);
```

Comment:
@joemag1 
Can you comment on that?
After reflecting that, I'm wondering, if that auto-ack was introduced in early development state of the tcp implementation the suppress retransmission. If so, we can remove it.
Usually the ACK on receiving the response should be enough in my opinion.